src/include/device.h

```diff
@@ -67,7 +67,12 @@ union ncclLLFifoLine {
   int4 i4;
 };
 
-#define WARP_SIZE warpSize
+#if defined(__GFX9__)
+#define WARP_SIZE 64
+#else
+#define WARP_SIZE 32
+#endif
```

Comment:
`WARP_SIZE` should *only* be defined for device compilation to avoid misuse.
```suggestion
#if __HIP_DEVICE_COMPILE__
#if defined(__GFX9__)
#define WARP_SIZE 64
#else
#define WARP_SIZE 32
#endif
#endif
```

Reply:
This somehow does not work as it seems like `__device__` function still gets compiled on host (just cant call from host). If we want this, we have to guard all our kernel with `if __HIP_DEVICE__COMPILE__` which sounds extreme.

In the current PR, `WARP_SIZE` will still be defined as 32 but it will not be used as I removed `__host__` decoration from funtions that reference it.